frontend/src/app/dataset/dataset.component.ts

```diff
@@ -60,11 +61,29 @@ export class DatasetComponent {
     this.plotRef = ref.first;
   }
 
+  /**
+   * Setter method to initialize the dataset with a reference to itself. This enables
+   * the dataset to self-destruct when needed.
+   * @param ref A reference to this component.
+   */
+  public setContainerRef(ref: ComponentRef<DatasetComponent>) {
+    this.containerRef = ref;
+  }
   /**
    * Triggered when toggle on page is clicked. Calls the plot component toggleTrace method.
    * @param channel The channel of the trace to toggle.
    */
   toggleTrace(channel) {
     this.plotRef.toggleTrace(this.ids.get(channel));
   }
+
+  /**
+   * Responsible for removing all traces from plot
+   * and removing self from datasets list.
```

Comment:
Simplify comment to:
Removes all traces from plot and removes self from dataset list.